src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.File.cs

```diff
@@ -174,7 +174,21 @@ public string GetPropertyValue(string propertyName)
             => Project.GetPropertyValue(propertyName);
 
         public void SetPropertyValue(string propertyName, string propertyValue)
-            => Project.SetProperty(propertyName, propertyValue);
+        {
+            var property = Project.GetProperty(propertyName);
+
+            if (property != null)
```

Comment:
Would the old implementation have set a not-yet-existent property? I haven't tested it out, but it seems like that API might have allowed setting properties that didn't exist whereas this one will treat those as a no-op.

Reply:
If `propertyValue is null` is the only time we should find the property. Otherwise, just remove it. It shouldn't be contingent on the property not existing, but on what the user passes in.

@mjrousos does that address it?

Reply:
Right, if `propertyValue` is null (or maybe empty), then we expect `property` to exist. If `propertyValue` exists, though, I'm not sure `propety` needs to.

I'm thinking this should look more like:

```CSharp
if (string.IsNullOrEmpty(propertyValue))
{
    var property = Project.GetProperty(propertyName);
    if (property != null)
    {
        Project.RemoveProperty(property);
    }
}
else
{
    Project.SetProperty(propertyName, propertyValue);
}
```

Reply:
It will Set or add a property with the specified name and value. I'll remove the check on property existing and separate out the remove functionality to make it cleaner.